downloader/src/main/kotlin/Downloader.kt

```diff
@@ -328,7 +329,7 @@ class Downloader(private val config: DownloaderConfiguration) {
                     gemDirectory.safeDeleteRecursively(force = true)
                 }
             } else {
-                sourceArchive.unpack(outputDirectory)
+                sourceArchive.unpack(outputDirectory, fallbackArchiveType = ArchiveType.TAR_GZIP)
```

Comment:
I don't feel comfortable with hard-coding this at this central place. I haven't looked at the details of the issue you're trying to address, but maybe we could instead set a correct file extension based on a mime-type in a HTTP response?

Reply:
I have some more ideas. Can we maybe have a look at this together?

Reply:
Sure. I have pushed my current state.